client/contentrouting.go

```diff
@@ -27,10 +28,28 @@ func (c *ContentRoutingClient) Provide(ctx context.Context, key cid.Cid, announc
 		return nil
 	}
 
-	_, err := c.client.Provide(ctx, key, 24*time.Hour)
+	_, err := c.client.Provide(ctx, []cid.Cid{key}, 24*time.Hour)
 	return err
 }
 
+func (c *ContentRoutingClient) ProvideMany(ctx context.Context, keys []multihash.Multihash) error {
+	keysAsCids := make([]cid.Cid, 0, len(keys))
+	for _, m := range keys {
+		keysAsCids = append(keysAsCids, cid.NewCidV0(m))
+	}
+	_, err := c.client.Provide(ctx, keysAsCids, 24*time.Hour)
+	return err
+}
+
+// Ready is part of the existing `ProvideMany` interface, but can be used more generally to determine if the routing client
+// has a working connection.
+func (c *ContentRoutingClient) Ready() bool {
```

Comment:
Sorry maybe I'm out of the loop here, but what is this for? What connection state are we envisioning using here? Checking if the underlying HTTP client has any conns? And how is this method going to be used by callers?

Reply:
i couldn't find documentation, but the current logic in kubo does an interface cast expecting this method to exist

Reply:
Ah okay thanks, I found it: https://github.com/ipfs/kubo/blob/master/routing/wrapper.go#L35

Looks like it's by the reprovider for the acclerated DHT client, to keep it from providing records until the routing table is bootstrapped.

Seems like a blocker for getting the accelerated DHT client working behind Reframe, but I think we can figure that out later.